crates/cext/src/parameter_expression.rs

```diff
@@ -0,0 +1,930 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashMap;
+use num_complex::Complex64;
+use std::ffi::{c_char, CStr, CString};
+use std::ptr;
+use std::sync::Arc;
+
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::operations::Param;
+use qiskit_circuit::parameter::parameter_expression::{ParameterError, ParameterExpression};
+use qiskit_circuit::parameter::symbol_expr::{Symbol, SymbolExpr, Value};
+
+/// @ingroup QkParam
+/// Construct a new Parameter with a name of symbol
+///
+/// @param name The name of symbol.
+///
+/// @return A pointer to the created ParameterExpression.
+///
+/// # Example
+///
+///     QkParam *a = qk_parameter_new("a");
+///
+/// # Safety
+///
+/// name must be a pointer to valid C string to name of symbol
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_parameter_new(name: *const c_char) -> *mut Param {
+    let name = unsafe { CStr::from_ptr(name).to_str().unwrap() };
```

Comment:
Every `unsafe` statement needs a safety comment explaining why it is safe to be called, e.g. here
```suggestion
    // SAFETY: Per documentation, ``name`` is a valid pointer to a C string
    let name = unsafe { CStr::from_ptr(name).to_str().unwrap() };
```